src/shogun/lib/parameter_observers/ParameterObserverCV.cpp

```diff
@@ -177,7 +165,13 @@ CrossValidationStorage* CParameterObserverCV::get_observation(int run) const
 	return obs;
 }
 
-const int CParameterObserverCV::get_num_observations() const
+const int32_t CParameterObserverCV::get_num_observations() const
 {
-	return m_observations.size();
+	try {
```

Comment:
do you need to try catch that here? We could just make the exception string in viktors code more verbose (printing the provided value and the ranges or something? Not sure I get why the error msg is here ...?

Reply:
This is in fact quite a bit neater. We should have thought of this in the first place :)

Reply:
Ok good, but this now deals with the issue properly which is always better than adding more and more hacks

Reply:
sooo is that try block okay? LoL

Reply:
Ah ok it is a warning, so just leave this, nevermind my comment :)